src/frontend/src/redux/search/selectors.ts

```diff
@@ -0,0 +1,25 @@
+import { createSelector } from "reselect";
+import { CaseData } from "../../components/RecordSearch/Record/types";
+import { AppState } from "../store";
+
+const getRecordState = (state: AppState) => state.search.record;
+
+export const hasBalanceDue = (element: CaseData) => {
+  return element.balance_due > 0;
+};
+
+export const selectCasesWithBalanceDue = createSelector(
+  [getRecordState],
+  (record) => {
+    if (record === undefined) {
+      return [];
+    }
+
+    const { cases } = record;
+    if (cases === undefined) {
```

Comment:
Since we're doing style refactors, it would be nice to be able to use the syntax `record?.cases?.filter(hasBalanceDue)` in a future PR.

Reply:
I went ahead and made this change now since it cleans up the implementation quite a bit.